### ISSUE TITLE
Tensorflow lite speech sample is not detecting silence (wrong detections)

### ISSUE THREAD

**Issue by mustansarsaeed** [tensorflow/tensorflow#43340]
Hi, I am using the sample https://github.com/tensorflow/examples/tree/master/lite/examples/speech_commands/android . I am facing the following issues:
- When there is a silence it detects as a speech?
- It also detects fan noise, clap or any tick noises as a speech which it should not as it is not a speech of a person.
- It does not detect voice correctly, like If I have spoken Yes it does not detect and same for other words as well.

Can you please guide which are the missing steps or which model to use if sample is not update? Speech and silence should be detected correctly. Currently it detects silence as words as well.

**Comment by wangtz** [comment]
Hi Mustansar,

The model is trained on small open dataset, contributed by the volunteers. If you want to improve the accuracy for background noise on fan noise and clap sounds, you can record some background noise dataset and use this training script to train a model with better quality.

https://www.tensorflow.org/tutorials/audio/simple_audio

Thanks,
Tiezhen

**Comment by saikumarchalla** [comment]
@mustansarsaeed  Please check the above comment and let us know if you still have any queries.
